test/tests/functional/pbs_hook_set_jobenv.py

```diff
@@ -885,3 +908,69 @@ def test_interactive_no_hook(self):
                          self.env_nohook['BROL1'].rstrip('\n'))
         self.logger.info("BROL - " + os.environ['BROL1'] + " == " +
                          self.env_nohook['BROL1'].rstrip('\n'))
+
+    def test_execjob_epi2(self):
+        """
+        Test that Variable_List will contain environment variable
+        with commas, newline and all special characters for a job
+        that has been recovered from a prematurely killed mom. This
+        is a test from an execjob_epilogue hook's view.
+        PRE: Set up currently executing user's environment to have variables
+             whose values have the special characters.
+             Submit a job using the -V option (pass current environment)
+             where there is an execjob_epilogue hook that references
+             Variable_List value.
+             Now kill -9 pbs_mom and then restart it.
+             This causes pbs_mom to read in job data from the *.JB file on
+             disk, and pbs_mom immediately kills the job causing
+             execjob_epilogue hook to execute.
+        POST: The epilogue hook should see the proper value to the
+              Variable_List.
+        """
+        j = Job()
+        a = {'Resource_List.select': '1:ncpus=1',
+             'Resource_List.walltime': 60}
+        j.set_attributes(a)
```

Comment:
This will be slightly shorter if you set a first and do Job(attrs=a)

Reply:
Good idea. I'll do this.